library/src/blas1/rocblas_iamaxmin_impl.h

```diff
@@ -0,0 +1,113 @@
+#pragma once
+
+#include "rocblas_iamaxmin_template.h"
+#include "rocblas_utils.h"
+
+//
+// Define names.
+//
+#define ROCBLAS_IAMAXMIN_NAME JOIN(rocblas_iamaxmin, JOIN(ROCBLAS_IAMAXMIN_GROUPKIND_SUFFIX, _name))
+
+template <typename>
+static constexpr char ROCBLAS_IAMAXMIN_NAME[] = "unknown";
+
+template <>
+static constexpr char ROCBLAS_IAMAXMIN_NAME<float>[]
+    = "rocblas_isa" QUOTE(MAX_MIN) QUOTE(ROCBLAS_IAMAXMIN_GROUPKIND_SUFFIX);
+
+template <>
+static constexpr char ROCBLAS_IAMAXMIN_NAME<double>[]
+    = "rocblas_ida" QUOTE(MAX_MIN) QUOTE(ROCBLAS_IAMAXMIN_GROUPKIND_SUFFIX);
+
+template <>
+static constexpr char ROCBLAS_IAMAXMIN_NAME<rocblas_float_complex>[]
+    = "rocblas_ica" QUOTE(MAX_MIN) QUOTE(ROCBLAS_IAMAXMIN_GROUPKIND_SUFFIX);
+
+template <>
+static constexpr char ROCBLAS_IAMAXMIN_NAME<rocblas_double_complex>[]
+    = "rocblas_iza" QUOTE(MAX_MIN) QUOTE(ROCBLAS_IAMAXMIN_GROUPKIND_SUFFIX);
+
+template <typename U>
+static rocblas_status rocblas_iamaxmin_impl(rocblas_handle handle,
+                                            rocblas_int    n,
+                                            U              x,
+                                            rocblas_int    incx,
+                                            rocblas_stride stridex,
+                                            rocblas_int    batch_count,
+                                            rocblas_int*   result)
+{
+    //
+    // Get the 'T input' type.
+    //
+    using Ti = batched_data_t<U>;
+
+    //
+    // Get the name of the routine.
+    //
+    static constexpr const char* const name = ROCBLAS_IAMAXMIN_NAME<Ti>;
+
+    //
+    // Get the 'T output' type
+    //
+    using To = typename reduction_types<Ti>::To;
+
+    //
+    // HIP support up to 1024 threads/work times per thread block/work group
+    //
+    static constexpr int NB = 1024;
+    if(!handle)
+    {
+        return rocblas_status_invalid_handle;
+    }
+
+    //
+    // Log trace.
+    //
+    rocblas_utils<U>::logging_trace(
+        handle, n, x, incx, stridex, batch_count, ROCBLAS_IAMAXMIN_NAME<Ti>);
+
+    //
+    // Log bench.
+    //
+    rocblas_utils<U>::logging_bench(handle,
+                                    n,
```

Comment:
What are the `logging_bench` and `logging_trace` functions?

The logging functions (in `logging.h`) were already designed to be general-purpose, taking an arbitrary number of arguments. These wrappers just add another layer of complexity, making them slightly harder to understand.

Only the `IAMAX`, `IAMIN`, `ASUM` and `NRM2` rocBLAS functions are similar enough to share logging code, and it's already been suggested that they be merged, where the logging can be done as well as other common functionality.

Reply:
I simplified